Source/Core/Core/State.cpp

```diff
@@ -388,32 +371,28 @@ static void CompressAndDumpState(CompressAndDumpState_args& save_args)
 
   if (header.size != 0)  // non-zero header size means the state is compressed
   {
-    lzo_uint i = 0;
+    u32 bytes_compressed = 0;
+
     while (true)
     {
-      lzo_uint32 cur_len = 0;
-      lzo_uint out_len = 0;
-
-      if ((i + IN_LEN) >= buffer_size)
-      {
-        cur_len = (lzo_uint32)(buffer_size - i);
-      }
-      else
-      {
-        cur_len = IN_LEN;
-      }
+      u32 bytes_to_compress = std::min((u32)LZ4_MAX_INPUT_SIZE, header.size - bytes_compressed);
```

Comment:
It's probably worth adding a utility 'compress buffer to file' function instead of having the compression logic inline like this. I know that's what the old code did but that code is ancient and we have different standards now.